src/main/java/frc/robot/util/Vision.java

```diff
@@ -127,17 +135,16 @@ public Pose3d getTagPose(int id){
     return getAprilTagFieldLayout().getTagPose(id).get();
   }
 
-  public void logging(){
-    //TODO: fix this, currently breaks because of nulls
-    // DoubleSupplier[] poseSupplier = {
-    //   () -> getPose2d().getX(),
-    //   () -> getPose2d().getY(),
-    //   () -> getPose2d().getRotation().getRadians()
-    // };
-    
-    // LogManager.addDoubleArray("Pose2d", poseSupplier);
-    // LogManager.addInt("Visible April Tags", () -> getEstimatedPoses(getPose2d()).size());
-    
+  public void logPose(Pose2d referencePose){
+    Pose2d pose = getPose2d(referencePose);
+    if(pose!=null){
+      DoubleSupplier[] poseSupplier = {
+        () -> pose.getX(),
+        () -> pose.getY(),
+        () -> pose.getRotation().getRadians()
+      };
+      LogManager.addDoubleArray("Pose2d", poseSupplier);
+    }
```

Comment:
this dosen't do anything but log one pose but as many entries and when you call it in robot container it will always be a (0,0,0) pose

Reply:
also the pose should prorobly be loged in drivetrain

Reply:
It used to log how many poses it was getting, but that doesn't help because the two cameras are always pointing the same direction, so it will almost always be 2. I did not write this, and whoever did thought it returned one pose for each April tag, which it doesn't. If you are using the comp bot, it will always say (0, 0) because the cameras are uncalibrated. This is vision pose, not robot pose, so it shouldn't be in Drivetrain.